api/v1alpha2/istio_merge.go

```diff
@@ -158,9 +159,74 @@ func (i *Istio) mergeConfig(op iopv1alpha1.IstioOperator) (iopv1alpha1.IstioOper
 		op.Spec.MeshConfig = updatedConfig
 	}
 
+	op = applyGatewayExternalTrafficPolicy(op, i)
+
 	return op, nil
 }
 
+func applyGatewayExternalTrafficPolicy(op iopv1alpha1.IstioOperator, i *Istio) iopv1alpha1.IstioOperator {
+	if i.Spec.Config.GatewayExternalTrafficPolicy != nil {
+		if op.Spec.Components == nil {
+			op.Spec.Components = &v1alpha1.IstioComponentSetSpec{}
+		}
+		if len(op.Spec.Components.IngressGateways) == 0 {
+			op.Spec.Components.IngressGateways = append(op.Spec.Components.IngressGateways, &v1alpha1.GatewaySpec{})
+		}
+		if op.Spec.Components.IngressGateways[0].K8S == nil {
+			op.Spec.Components.IngressGateways[0].K8S = &v1alpha1.KubernetesResourcesSpec{}
+		}
+
+		const kind = "Service"
+		const version = "v1"
+		const istioIngressGateway = "istio-ingressgateway"
+		const path = "spec.externalTrafficPolicy"
+		const topologyKey = "kubernetes.io/hostname"
+		const app = "app"
+
+		op.Spec.Components.IngressGateways[0].K8S.Overlays = append(op.Spec.Components.IngressGateways[0].K8S.Overlays, &v1alpha1.K8SObjectOverlay{
+			ApiVersion: version,
+			Kind:       kind,
+			Name:       istioIngressGateway,
+			Patches: []*v1alpha1.K8SObjectOverlay_PathValue{
+				{
+					Path:  path,
+					Value: structpb.NewStringValue(*i.Spec.Config.GatewayExternalTrafficPolicy),
+				},
+			},
+		})
+
+		if *i.Spec.Config.GatewayExternalTrafficPolicy == "Local" {
+			op.Spec.Components.IngressGateways[0].K8S.Affinity = &v1alpha1.Affinity{
+				PodAntiAffinity: &v1alpha1.PodAntiAffinity{
```

Comment:
I was thinking about using [topologySpreadConstraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) instead of PodAntiAffinity. With topologySpreadConstraints you can configure `maxSkew` what might give better control over scheduling pods equally in nodes. I see a different number of pods on the nodes as the main risk for reduced performance when using externalTrafficPolicy Local.

What do you think?

Reply:
I was also thinking about `topologySpreadConstraints` as it seems to be the generally better solution, but, the problem with `topologySpreadConstraints` is that it is not supported in IstioOperator config, and would require that we use `overlay` instead of providing it as configuration for a field supported in IOP by default.

I don't see a big issue with staying with `podAntiAffinity` as the scheduler should distribute the load equally between nodes, meaning that we should expect the same outcome as configuring `maxSkew` to 1. (which I saw while testing)

I would want to avoid using `overlay` if possible, as this might cause issues with the field not being cleared up from the deployment after change Local -> Cluster (istio install merge policy). 

And, as the outcome of running performance tests multiple times, it doesn't look like there should be a high impact anyway, but if we observe issue we can think about adjusting it.

Reply:
Wait, we have podAntiAffinity in Istio Operator by default !?
https://github.com/kyma-project/istio/blob/a5b62d61626f77a7cea0e00d2dc6af3d3c6eb556/internal/istiooperator/istio-operator.yaml#L19
I somehow missed this...

Reply:
Oh good finding ... I also missed this. 🙈 

I agree with what you said about `topologySpreadConstraints`. 👍